llvm/lib/CodeGen/MachineBasicBlock.cpp

```diff
@@ -1781,9 +1781,6 @@ MachineBasicBlock::livein_iterator MachineBasicBlock::livein_begin() const {
 
 MachineBasicBlock::liveout_iterator MachineBasicBlock::liveout_begin() const {
   const MachineFunction &MF = *getParent();
-  assert(MF.getProperties().hasTracksLiveness() &&
-         "Liveness information is accurate");
-
```

Comment:
Note: This assert occurs too early (so asserts for blocks with no successors), removing this is safe as the `liveout_iterator` internally calls `livein_begin()`, which contains the same assert.